ttl.go

```diff
@@ -121,29 +128,50 @@ func (m *expirationMap[V]) cleanup(store store[V], policy policy[V], onEvict fun
 
 	m.Lock()
 	now := time.Now()
-	bucketNum := cleanupBucket(now)
-	keys := m.buckets[bucketNum]
-	delete(m.buckets, bucketNum)
+	currentBucketNum := cleanupBucket(now)
+	// Clean up all buckets up to and including currentBucketNum, starting from
+	// (but not including) the last one that was cleaned up
+	var buckets []bucket
+	for bucketNum := m.lastCleanedBucketNum + 1; bucketNum <= currentBucketNum; bucketNum++ {
+		buckets = append(buckets, m.buckets[bucketNum])
+		delete(m.buckets, bucketNum)
+	}
+	m.lastCleanedBucketNum = currentBucketNum
 	m.Unlock()
 
-	for key, conflict := range keys {
-		expr := store.Expiration(key)
-		// Sanity check. Verify that the store agrees that this key is expired.
-		if expr.After(now) {
-			continue
+	for _, keys := range buckets {
+		for key, conflict := range keys {
+			expr := store.Expiration(key)
+			// Sanity check. Verify that the store agrees that this key is expired.
+			if store.Expiration(key).After(now) {
```

Comment:
Do we need to call `store.Expiration` here again?

Reply:
No that's a mistake, can just compare to expr

Reply:
I merged this PR and created another PR to fix this. Thanks for your contributions.